File: library/src/main/java/io/github/anderscheow/validator/util/ErrorMessage.java
package io.github.anderscheow.validator.util;

import android.support.annotation.NonNull;
import android.support.annotation.StringRes;

public abstract class ErrorMessage {

    private int errorRes = -1;

    private String errorMessage = "Invalid input";

    @StringRes
    public int errorRes() {
        return errorRes;
    }

    @NonNull
    public String errorMessage() {
        return errorMessage;
    }

    public void setErrorRes(@StringRes int errorRes) {
        this.errorRes = errorRes;
    }

    public void setErrorMessage(@NonNull String errorMessage) {
        this.errorMessage = errorMessage;
    }

    public boolean isErrorAvailable() {
        return isErrorResAvailable() || isErrorMessageAvailable();
    }

    public boolean isErrorResAvailable() {
        return errorRes() != -1;
    }

    public boolean isErrorMessageAvailable() {
        return !errorMessage().isEmpty();
    }
}

File: library/src/main/java/io/github/anderscheow/validator/util/ErrorMessage.kt
package io.github.anderscheow.validator.util

import android.support.annotation.StringRes

interface ErrorMessage {

    val isErrorAvailable: Boolean
        get() = isErrorResAvailable || isErrorMessageAvailable

    val isErrorResAvailable: Boolean
        get() = getErrorRes() != -1

    val isErrorMessageAvailable: Boolean
        get() = !getErrorMessage().isEmpty()

    @StringRes
    fun getErrorRes(): Int

    fun getErrorMessage(): String
}
